src/oracles/ChainlinkPriceOracle.sol

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity >=0.8.0 <0.9.0;
+
+import {AggregatorV3Interface} from "../interfaces/AggregatorV3Interface.sol";
+import {IPriceOracle} from "../interfaces/IPriceOracle.sol";
+import {IWatchtowerCustomErrors} from "../interfaces/IWatchtowerCustomErrors.sol";
+/**
+ * @title Chainlink Price Oracle
+ * @author GUNBOATs
+ * @dev This contract creates an oracle that is compatible with the IPriceOracle
+ * interface and can be used by a CoW AMM to determine the current price of the
+ * traded tokens from Chainlink Data Feeds with decimals lower than 18.
+ */
+
+contract ChainlinkPriceOracle is IPriceOracle, IWatchtowerCustomErrors {
+    /**
+     * @param token0Feed Address of token0 oracle
+     * @param token1Feed Address of token1 oracle
+     * @param timeThreshold Amount of seconds before it consider as stale oracle
+     * @param backoff Duration to indicated how long the watchtower will wait for oracle to refresh
+     */
+    struct Data {
+        address token0Feed;
+        address token1Feed;
+        uint256 timeThreshold;
+        uint256 backoff;
+    }
+    /**
+     * @dev There is no mapping between Chainlink oracle and the token address,
+     * so the user must check the that the oracle is corresponds to the token themselves.
+     * @inheritdoc IPriceOracle
+     */
+
+    function getPrice(address, address, bytes calldata data)
+        external
+        view
+        returns (uint256 priceNumerator, uint256 priceDenominator)
+    {
+        Data memory OracleData = abi.decode(data, (Data));
+        AggregatorV3Interface token0Feed = AggregatorV3Interface(OracleData.token0Feed);
+        AggregatorV3Interface token1Feed = AggregatorV3Interface(OracleData.token1Feed);
+        (
+            /* uint80 roundId*/
+            ,
```

Comment:
Is this how you tell the compiler that some variables are not used?
Maybe moving the commented out variable and the comma in the same line would be nicer to read?
```
/* uint80 roundId*/ ,
```

Reply:
this is how `forge fmt` format it right now.

Reply:
Bummer, looks pretty weird to me but if this the standard we should stick with that formatting. 👌